### PR TITLE
Rework ConvertToProjectWithPomFileTest test for checking conversion folders to maven multimodule project

### PR DESCRIPTION
**Pull request by musienko-maxim**
### What does this PR do?
* Implement scenario which was described in the issue: https://github.com/eclipse/che/issues/6905
* Fix getProject method (add authorize header)  for correct working on multiuser/OpenShift assemblies
* Fix template for maven project (`artifactId` was the same for parent and module and this produces conflict)
* Add `undefined` project type for checking autosetting type for projects like this

### What issues does this PR fix or reference?
 https://github.com/eclipse/che/issues/6905

### CODE REVIEW

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.miscellaneous;
+
+import static java.lang.String.*;
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: As far as I know, the common import is a bad practice. Please import relations directly.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.InformationDialog;
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Aleksandr Shmaraev */
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: Is the author name correct?

**Review by SkorikSergey — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    ide.waitOpenedWorkspaceIsReadyToUse();
+    projectExplorer.quickRevealToItemWithJavaScript(format("%s/%s", PROJECT_NAME, WEB_APP_MODULE));
+  }
+
+  @Test
+  public void shouldConvertToMavenMultimoduleProject() throws Exception {
+    convertPredefinedFolderToMavenProjectWithContextMenu(
+        format("%s/%s", PROJECT_NAME, WEB_APP_MODULE));
+    testProjectServiceClient.checkProjectType(
+        workspaceId, format("%s/%s", PROJECT_NAME, WEB_APP_MODULE), "maven");
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: What about move all these items to the upper of the test and declare them as constants.
```
final String convertPath = format("%s/%s", PROJECT_NAME, WEB_APP_MODULE);
```

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    addParentConfigurationToPredefinedFolder();
+    menu.runCommand(
+        TestMenuCommandsConstants.Project.PROJECT,
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: Static import?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    convertToMavenByhWizard("/", PROJECT_NAME);
+    testProjectServiceClient.checkProjectType(
+        workspaceId, format("%s/%s", PROJECT_NAME, WEB_APP_MODULE), "maven");
+  }
+
+  @Test
+  public void shouldNotConvertToMavenProject() {
+    projectExplorer.waitAndSelectItem(NONE_MAVEN_PROJECT);
+    menu.runCommand(
+        TestMenuCommandsConstants.Project.PROJECT,
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: Static import?

**Review by Ohrimenko1988 — APPROVED** [review]

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java [hunk 6]
```diff
@@ -293,6 +293,7 @@ private ProjectConfig getProject(String workspaceId, String projectName) throws
     return requestFactory
         .fromUrl(workspaceAgentApiEndpointUrlProvider.get(workspaceId) + "project/" + projectName)
         .useGetMethod()
+        .setAuthorizationHeader(machineServiceClient.getMachineApiToken(workspaceId))
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java]: Should there be added prefix `BEARER_TOKEN_PREFIX + ` as well?

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/project/ProjectTemplates.java [hunk 7]
```diff
@@ -25,6 +25,7 @@
   public static final String CONSOLE_JAVA_SIMPLE = "console_java_simple.json";
   public static final String GO = "go.json";
   public static final String DOT_NET = "dotNet.json";
+  public static final String UNDEFINED_PROJECT = "undefined.json";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/project/ProjectTemplates.java]: IMHO next name would better reflect template proposal: `PROJECT_OF_UNDEFINED_TYPE`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  @Inject private Menu              menu;
+  @Inject private Loader            loader;
+  @Inject private Wizard            wizard;
+  @Inject private AskForValueDialog askForValueDialog;
+  @Inject private InformationDialog informationDialog;
+  @Inject private ActionsFactory           actionsFactory;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  private String workspaceId;
+  @BeforeClass
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: Please, add empty line to improve readability.

Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: Done

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  private void convertPredefinedFolderToMavenProjectWithContextMenu(String converPath) {
+    projectExplorer.waitAndSelectItem(converPath);
+    projectExplorer.openContextMenuByPathSelectedItem(converPath);
+    projectExplorer.clickOnItemInContextMenu(CONVERT_TO_PROJECT);
+    convertToMavenByhWizard("/" + PROJECT_NAME, WEB_APP_MODULE);
+  }
+
+  private void convertToMavenByhWizard(String pathToDirectory, String convertedFolder) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: typo `convertToMavenByhWizard`
Also, this method could be useful for other tests, and would be better moved into the Wizard page object.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Aleksandr Shmaraev, Musienko Maxim */
+public class CheckConvertingToMavenProjectTest {
+  private static final String PROJECT_NAME = NameGenerator.generate("project", 4);
+  private static final String WEB_APP_MODULE = "my-webapp";
+  private static final String NONE_MAVEN_PROJECT = NameGenerator.generate("noneMavenProject", 4);
+  private static final String PARENT_INFORMATION =
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: I would suggest to use more specific name `PARENT_ARTIFACT_SECTION`.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+    wizard.waitOpenProjectConfigForm();
+    wizard.waitTextParentDirectoryName(pathToDirectory);
+    wizard.waitTextProjectNameInput(convertedFolder);
+    wizard.selectSample(Wizard.TypeProject.MAVEN);
+    loader.waitOnClosed();
+    wizard.clickSaveButton();
+    wizard.waitCloseProjectConfigForm();
+  }
+
+  private void addParentConfigurationToPredefinedFolder() {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: I would suggest to use more specific name `addParentArticatSectionIntoPomFile`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+import org.eclipse.che.selenium.pageobject.InformationDialog;
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.Wizard;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Aleksandr Shmaraev, Musienko Maxim */
+public class CheckConvertingToMavenProjectTest {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/miscellaneous/CheckConvertingToMavenProjectTest.java]: Test name could be simplified slightly: `ConvertToMavenProjectTest`

**Review by dmytro-ndp — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
ci-build

**Comment by musienko-maxim** [comment]
ci-build

**Pull request merged**